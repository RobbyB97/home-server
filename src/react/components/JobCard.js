import React from 'react'

export default (props) => (
    <div className="jobcard">
        {props.title && 
            <h2 className="jobcard__title">
                {props.title}
            </h2>
        }

        {props.text &&
            <p className="jobcard__text">
                {props.text}
            </p>
        }

        {props.image &&
            <img className="jobcard__image"
                src={props.image}/>
        }

    </div>
)

//export default JobCard