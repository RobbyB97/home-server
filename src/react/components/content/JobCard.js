import React from 'react'

export default (props) => (
    <div className="jobcard">
        {props.title && 
            <h2 className="jobcard__title">
                {props.title}
            </h2>
        }

        {props.text &&
            <p className="jobcard__text">
                {props.text}
            </p>
        }

        {props.image &&
            <div className="jobcard__image">
                <img src={props.image} />
            </div>
        }

    </div>
)
