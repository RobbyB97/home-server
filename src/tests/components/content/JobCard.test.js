import React from 'react'
import {shallow} from 'enzyme'

import JobCard from '../../../react/components/content/JobCard'


let wrapper

beforeEach(() => {
    wrapper = shallow(
        <JobCard 
            image="Image!"
            title="Title!"
            text="Text!"
        />
    )
})


test('Render JobCard with no props', () => {
    wrapper = shallow(<JobCard />)

    expect(wrapper).toMatchSnapshot()
})


test('Render JobCard with props', () => {
    wrapper = shallow(
        <JobCard 
            image="Image!"
            title="Title!"
            text="Text!"
        />
    )

    expect(wrapper).toMatchSnapshot()
})