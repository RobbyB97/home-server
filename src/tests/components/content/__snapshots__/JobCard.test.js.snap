// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render JobCard with no props 1`] = `
<div
  className="jobcard"
/>
`;

exports[`Render JobCard with props 1`] = `
<div
  className="jobcard"
>
  <h2
    className="jobcard__title"
  >
    Title!
  </h2>
  <p
    className="jobcard__text"
  >
    Text!
  </p>
  <div
    className="jobcard__image"
  >
    <img
      src="Image!"
    />
  </div>
</div>
`;
